### ISSUE TITLE
[fr] incosistent apostrophe suggestion

### ISSUE THREAD

**Issue by tiff** [languagetool-org/languagetool#5239]
> Il mattendait
> Cest la vie

The first one suggests `m'attendait` and the second one suggests `C’est` (with a typographic apostrophe). We should only suggest one type of apostrophe.

**Comment by udomai** [comment]
Note to self: evaluate different cases – do we get all of them?

Example with no good correction: _Je crois que personne ne pouvait **sattendre** à ça._ (ideal correction: s'attendre; actual suggestion: attendre, s attendre)

**Comment by jaumeortola** [comment]
> The first one suggests m'attendait and the second one suggests C’est (with a typographic apostrophe). We should only suggest one type of apostrophe.

Yes. You are right. Internally, everything should be written with straight apostrophes (dictionaries, rules, suggestions...). 

If the picky rule that requires typographical apostrophe is enabled, then the suggestions should use typographical apostrophes. This is implemented for Catalan and LibreOffice. It should be done for all languages on picky mode. See also: https://github.com/languagetooler-gmbh/languagetool-premium/issues/2639

**Comment by jaumeortola** [comment]
> This is implemented for Catalan and LibreOffice.

See: https://github.com/languagetool-org/languagetool/blob/7c908aeabd1275de1e9fd418df5b9c85d389346c/languagetool-core/src/main/java/org/languagetool/rules/LanguageDependentFilter.java#L47

**Comment by jaumeortola** [comment]
Now, in French and Catalan picky mode, the apostrophes in suggestions will be typographical. There is a picky rule that requires typographical apostrophes, and it is inconsistent to suggest something (a straight apostrophe) that will be changed by another rule. https://github.com/languagetool-org/languagetool/commit/c596901dc85759de20b9e700c81a9d744ef81894

This feature can be added to other languages (English, German...), but it doesn't seem important. In English, the rules for typographical apostrophes or quotes are now disabled even in picky mode.

**Comment by jaumeortola** [comment]
This is not working as I expected (typographical apostrophes on picky mode, and straight apostrophes on non-picky) in French and Catalan suggestions. We are giving all suggestions with typographical apostrophes. The reason is that the web always asks the server the full results, including the picky rules. 

This has created other problems in the past. Perhaps we need some "mixed mode" (that means "give me the picky mode results, but we are not really on picky mode and we are not going to show them"). @tiff  

We can leave it as it is now. The only full solution would be a user setting for typographical/typewriter apostrophes.
